lib/open_api_spex/schema.ex

```diff
@@ -291,6 +292,9 @@ defmodule OpenApiSpex.Schema do
 
     - Cast the value using each schema listed in `anyOf`, stopping as soon as a succesful cast is made.
   """
+  def cast(%Schema{"x-validate": module} = schema, value, schemas) when module != nil,
```

Comment:
The error return value of `Cast.cast/3` is incompatible with the error return value of `Schema.cast/3`.

`Schema.cast/3` is slated to be deprecated soon. `OpenApiSpex.Cast.cast/3` replaces it.

Let's remove this function head and move its test to `CastTest`.